forui/lib/src/theme/breakpoints.dart

```diff
@@ -0,0 +1,146 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+/// The responsive breakpoints based on [Tailwind CSS](https://tailwindcss.com/docs/responsive-design).
+///
+/// All breakpoints are in logical pixels. It is not recommended to change the default breakpoints unless
+/// you absolutely know what you are doing.
+///
+/// Usage:
+/// ```dart
+/// @override
+/// Widget build(BuildContext context) {
+///   final theme = context.theme; // FThemeData
+///   final size = MediaQuery.sizeOf(context); // Size
+///   if (theme.breakpoints.sm.contains(size)) {
+///     // Do something when the sm breakpoint is active.
+///     return SizedBox.square(dimension: 100);
+///
+///   } else {
+///     return SizedBox.square(dimension: 500);
+///   }
+/// }
+/// ```
+///
+/// Additional breakpoints can be added via an extension on this class:
+/// ```dart
+/// extension CustomBreakpoints on FBreakpoints {
+///   FBreakpoint get custom => const FBreakpoint(min: 100, max: 200);
+/// }
+/// ```
+///
+/// After which, the custom breakpoint can be accessed like so:
+/// ```dart {5}
+/// @override
+/// Widget build(BuildContext context) {
+///   final theme = context.theme; // FThemeData
+///   final size = MediaQuery.sizeOf(context); // Size
+///   if (theme.breakpoints.custom.contains(size)) {
+///     // Do something when the custom breakpoint is active.
+///     return SizedBox.square(dimension: 100);
+///
+///   } else {
+///     return SizedBox.square(dimension: 500);
+///   }
+/// }
+/// ```
+final class FBreakpoints with Diagnosticable {
+  /// The extra small breakpoint. Defaults to `0 <= width < 640`.
+  final FBreakpoint xs;
+
+  /// The small breakpoint. Defaults to `640 < width < 768`.
+  final FBreakpoint sm;
+
+  /// The medium breakpoint. Defaults to `768 < width < 1024`.
+  final FBreakpoint md;
+
+  /// The large breakpoint. Defaults to `1024 < width < 1280`.
+  final FBreakpoint lg;
+
+  /// The extra large breakpoint. Defaults to `1280 < width < 1536`.
+  final FBreakpoint xl;
+
+  /// The extra extra large breakpoint. Defaults to `1536 < width`.
+  final FBreakpoint xl2;
+
+  /// Creates a [FBreakpoints].
+  const FBreakpoints({
+    this.xs = const FBreakpoint(max: 640),
+    this.sm = const FBreakpoint(min: 640, max: 768),
+    this.md = const FBreakpoint(min: 768, max: 1024),
+    this.lg = const FBreakpoint(min: 1024, max: 1280),
+    this.xl = const FBreakpoint(min: 1280, max: 1536),
+    this.xl2 = const FBreakpoint(min: 1536),
```

Comment:
Is it a potential issue if the user misses out on a particular breakpoint? For example, if the user needs breakpoints at xs and lg. This implementation essentially forces the user to implement, xs, sm, md, lg if they would like to scale up to lg.

The tailwind one makes sense since it only refers to the min-width and not the range of the width.